File: app/src/main/java/gateway/hitrontech/com/encryption/bean/EncryptionBean.java
package gateway.hitrontech.com.encryption.bean;

import static android.content.ContentValues.TAG;

import android.databinding.BaseObservable;
import android.databinding.Bindable;
import android.util.Log;
import gateway.hitrontech.com.encryption.BR;
import gateway.hitrontech.com.encryption.utils.Constants;

public class EncryptionBean extends BaseObservable {

  private String plainText = "";

  private String cipherText = "";


  private String key = Constants.KEY;

  @Bindable
  public String getPlainText() {
    return plainText;
  }

  public void setPlainText(String plainText) {

    if (!this.plainText.equals(plainText)) {
      this.plainText = plainText;
      notifyPropertyChanged(BR.plainText);
    }
  }

  @Bindable
  public String getCipherText() {
    return cipherText;
  }

  public void setCipherText(String cipherText) {
    if (!this.cipherText.equals(cipherText)) {
      this.cipherText = cipherText;
      notifyPropertyChanged(BR.cipherText);
    }
  }

  @Bindable
  public String getKey() {
    return key;
  }

  public void setKey(String key) {
    if (!this.key.equals(key)) {
      this.key = key;
      Log.e(TAG, "setKey has been invoked" + this.key);
      notifyPropertyChanged(BR.key);
    }
  }

  @Override
  public boolean equals(Object obj) {

    if (obj instanceof EncryptionBean) {
      return ((EncryptionBean) obj).plainText.equals(this.plainText) &&
          ((EncryptionBean) obj).key.equals(this.key);
    } else {
      return false;
    }
  }
}

File: app/src/main/java/gateway/hitrontech/com/encryption/bean/EncryptionBean.kt
package gateway.hitrontech.com.encryption.bean

import android.databinding.BaseObservable
import android.databinding.Bindable
import android.os.Parcel
import android.os.Parcelable
import gateway.hitrontech.com.encryption.BR
import gateway.hitrontech.com.encryption.utils.Constants
import java.io.Serializable

class EncryptionBean() : BaseObservable(), Serializable, Parcelable {


    @get:Bindable
    var plainText: String = String()
        set(plainText) {

            if (field != plainText) {
                field = plainText
                notifyPropertyChanged(BR.plainText)
            }
        }

    @get:Bindable
    var cipherText: String = String()
        set(cipherText) {
            if (field != cipherText) {
                field = cipherText
                notifyPropertyChanged(BR.cipherText)
            }
        }

    @get:Bindable
    var key: String = Constants.KEY
        set(key) {
            if (field != key) {
                field = key
                notifyPropertyChanged(BR.key)
            }
        }

    constructor(parcel: Parcel) : this() {
        plainText = parcel.readString()!!
        cipherText = parcel.readString()!!
        key = parcel.readString()!!
    }

    override fun equals(other: Any?): Boolean {
        var isEqual = false
        if (other is EncryptionBean) {
            if (other.plainText == plainText && other.key == key) {
                isEqual = true
            }
        }

        return isEqual
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {

    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<EncryptionBean> {
        override fun createFromParcel(parcel: Parcel): EncryptionBean {
            return EncryptionBean(parcel)
        }

        override fun newArray(size: Int): Array<EncryptionBean?> {
            return arrayOfNulls(size)
        }
    }
}
